.github/workflows/default.yaml

```
@@ -109,10 +114,30 @@ jobs:
           sudo apt-get --quiet=2 build-dep squid
           sudo apt-get --quiet=2 install linuxdoc-tools libtool-bin
 
+      - name: Install prerequisite MacOS packages
+        if: runner.os == 'macOS'
+        run: |
+          brew install \
+            automake coreutils cppunit gawk \
+            gnu-getopt gnu-sed grep \
+            make openldap cyrus-sasl
+
       - name: Checkout sources
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
-      - run: ./test-builds.sh
+      - name: Run build on Linux
+        if: runner.os == 'Linux'
+        run: ./test-builds.sh --aggressively-use-config-cache ${{ matrix.layer }}
+
+      - name: Run build on MacOS
+        if: runner.os == 'macOS'
+        run: |
+          eval `brew shellenv`
+          export PKG_CONFIG_PATH="$HOMEBREW_PREFIX/opt/openldap/lib/pkgconfig:$HOMEBREW_PREFIX/opt/cyrus-sasl/lib/pkgconfig"
+          export GETOPT="$HOMEBREW_PREFIX/opt/gnu-getopt/bin/getopt"
+          export pjobs="-j`nproc`"
+          export MAKE="$HOMEBREW_PREFIX/bin/gmake ${pjobs}"
```

I assume that `-j` is an optimization rather than a requirement to build on MacOS. Let's not increase the difference between Linux and MacOS builds in this PR where we do not have to:

```suggestion
          export MAKE="$HOMEBREW_PREFIX/bin/gmake"
```

If my assumption is wrong, please add a (YAML) comment to document what is going on in this special case.